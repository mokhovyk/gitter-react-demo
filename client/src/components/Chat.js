// @flow
import React, { Fragment } from 'react';

type PropsT = {
  messages: Array<any>,
  isLoading: boolean,
  name: string,
  onLoadMore: () => void,
  hasMore: boolean
};

const Chat = ({ messages, isLoading, name, onLoadMore, hasMore }: PropsT) => (
  <Fragment>
    <h2>{name}</h2>
    {messages && messages.length !== 0 &&
      <div className="flex-colum mb-3">
        {messages.map((item: any) => (
          <div key={item.id} className="py-2 border-bottom">
            <div className="text-muted">{item.fromUser.displayName}</div>
            <div dangerouslySetInnerHTML={{ __html: item.html }} />
          </div>
        ))}
      </div>
    }
    {isLoading && <div>Loading...</div>}
    {!isLoading && hasMore && <button onClick={onLoadMore} className="btn btn-primary">Load more</button>}
  </Fragment>
);

export default Chat;
