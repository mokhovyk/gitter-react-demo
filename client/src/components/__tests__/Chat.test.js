import React from 'react';
import Chat from '../Chat';

let props;
let component;

describe('Chat', () => {
  beforeEach(() => {
    props = {
      onClick: jest.fn(),
      messages: [],
      isLoading: false,
      name: 'My Chat',
      onLoadMore: jest.fn(),
      hasMore: true,
    };
    component = shallow(
      <Chat {...props} />,
    );
  });

  test('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  test('renders correctly without load more button', () => {
    component = shallow(
      <Chat
        {...props}
        hasMore={false}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  test('click correctly', () => {
    component.find('button').simulate('click', { preventDefault: jest.fn() });
    expect(props.onLoadMore).toBeCalled();
  });
});
