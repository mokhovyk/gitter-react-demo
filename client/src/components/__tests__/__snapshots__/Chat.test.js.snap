// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat renders correctly 1`] = `
<React.Fragment>
  <h2>
    My Chat
  </h2>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Load more
  </button>
</React.Fragment>
`;

exports[`Chat renders correctly without load more button 1`] = `
<React.Fragment>
  <h2>
    My Chat
  </h2>
</React.Fragment>
`;
